src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs

```diff
@@ -217,5 +218,32 @@ void M1()
                 "using System;\r\n" + arrayEmptySource + fixedSource.Replace("System.Array.Empty", "Array.Empty"),
                 allowNewCompilerDiagnostics: true);
         }
+
+        [WorkItem(10214, "https://github.com/dotnet/roslyn/issues/10214")]
+        [Fact(Skip = "10214")]
+        public void EmptyArrayVisualBasic_CompilerGeneratedArrayCreation()
+        {
+            const string arrayEmptySourceRaw = @"
+Namespace System
+    Public Class Array
+       Public Shared Function Empty(Of T)() As T()
+           Return Nothing
+       End Function
+    End Class
+End Namespace
+";
+            const string source = @"
+Class C
+    Sub M1()
+        Dim a = ""STR "".TrimEnd()               ' Compiler seems to generate a param array with size 0 for the TrimEnd invocation.
+    End Sub
+End Class";
+
+            string arrayEmptySource = IsArrayEmptyDefined() ? string.Empty : arrayEmptySourceRaw;
+
```

Comment:
Ok, I'm confused. Are you saying that in your sample code, the compiler would generate an empty array even on platforms where `Array.Empty` does not exist?